cpu/cc2538/periph/cpuid.c

```diff
@@ -1,45 +0,0 @@
-/*
- * Copyright (C) 2014 Loci Controls Inc.
- *
- * This file is subject to the terms and conditions of the GNU Lesser General
- * Public License v2.1. See the file LICENSE in the top level directory for more
- * details.
- */
-
-/**
- * @ingroup     cpu_cc2538
- * @{
- *
- * @file
- * @brief       CPU-ID driver implementation
- *
- * The CC2538 provides a 64-bit unique identifier, that is unique for each shipped unit.
- *
- * @author      Ian Martin <ian@locicontrols.com>
- *
- * @}
- */
-
-#include <stdint.h>
-#include <string.h>
-
-#include "cpu.h"
-#include "periph/cpuid.h"
-
-void cpuid_get(void *id)
-{
-    uint8_t *dest = (uint8_t *)id;
-
-    /**
-     * The byte-order is big-endian but the word order is little endian.
-     * Make some sense of it:
-     */
-    dest[0] = IEEE_ADDR_MSWORD >> (3 * sizeof(uint8_t));
```

Comment:
I'm fairly convinced this was completely wrong in the original implementation, the sizeof() is in bytes, but the shift should be in number of bits.